core/src/main/scala/chisel3/RawModule.scala

```diff
@@ -276,10 +277,10 @@ abstract class RawModule extends BaseModule {
 
 /** Enforce that the Module.reset be Asynchronous (AsyncReset) */
 trait RequireAsyncReset extends Module {
-  override final def resetType = Module.ResetType.Asynchronous
+  override final def resetType: Module.ResetType.Type = Module.ResetType.Asynchronous
```

Comment:
is this really what you want here? I would think that we *do* want to tighten the type here. Same with below

Reply:
It's debatable, I could go either way but I think you're probably right that the more concrete type is better here.